src/core/qgsvectorfilewriter.cpp

```diff
@@ -2446,6 +2446,7 @@ class QgsVectorFileWriterMetadataContainer
                              QObject::tr( "Compression method." ),
                              QStringList()
                              << QStringLiteral( "UNCOMPRESSED" )
+                             << QStringLiteral( "ZSTD" )
```

Comment:
Not all builds of libparquet have necessarily support for ZSTD conpression. It might be worth checking with something like:
```
GDALDriverH hParquetDrv = GDALGetDriverByName("PARQUET");
if (hParquetDrv)
    bHasZSTD = strstr(GDALGetMetadataItem(hParquetDrv, GDAL_DS_LAYER_CREATIONOPTIONLIST, nullptr), "ZSTD") != nullptr;
```